### DISCUSSION TITLE
Ability to do DRIP

### DISCUSSION THREAD

**Discussion by greenship24** [ghostfolio/ghostfolio#1705] [Ideas]
Is there an ability to do a DRIP transaction? Or will it be added?

To clarify, I am looking for a dividend reinvestment program kind of transaction where a security pays out a dividend and the payout is immediately reinvested back into the same security.

I can select dividend in the dropdown of course but it appears this does not actually function as a DRIP style of transaction but more of one that paid out at one point in time. To fully be a DRIP transaction I would have to not only add the dividend but also another buy transaction.

**Comment by dtslvr** [comment]
Hi @greenship24

A dedicated DRIP transaction is currently not supported in Ghostfolio.

Could you share a sample CSV export of your brokerage's data, specifically showing how they represent DRIP transactions? This would allow us to analyze the format for a potential implementation.

Thank you in advance.

**Comment by greenship24** [comment]
### Situation
For my brokerage, etc. it appears differently based on what you look at. For the CSV, it shows as you see with 0 shares bought, "1" as the share price, then the net amount you receive as the dividend. In the PDF statement, it shows "-" for everything but the Amount which is just the cash value. ($4.49 in my example)

A secondary transaction is made just as a BUY essentially with the share price that day, the net amount from the dividend, and then how many shares were able to be bought at that price.


Attached is the CSV showing what I mean. I had thought the "dividend" portion in Ghostfolio was doing DRIP only because I'm used to double entry accounting where that dividend would go OUT the brokerage account but IN to another account (like a checking account or money market or whichever direction you'd send the money.) I am only getting into Ghostfolio but I don't think it accounts for money going out unless it's to another account you keep?

### Problem
That is all fine of course. Just trying to understand and quite frankly, I don't need double entry here- I just don't see how a dividend transaction can apply here faithfully (ie. accurately for your accounts). Llooking at my charts, they don't match the slide as it should be. Ie. I think I'm down 20% in Ghostfolio yet the reality is just under 10%. Trying to lop off dividends and just use "BUY" statements doesn't work either as my portfolio shows more invested than I have actually (ie. no dividends) so it still remains 20%.

Summing up the previous paragraph, using a DIVIDEND and a BUY transaction will not be right (ie. more money invested than you actually did, thus distorting your performance) and using only a BUY has the same issue- you didn't "buy" it with your money- your fund grew and then gave you some earnings. Same case applies to long term and short term capital gains which are reinvested.

### Prescription
So in this case, I think ideally a DRIP transaction should appear as such:

Type: DRIP
Account: X
Name, symbol: VOO
Date
Quantity: quantity of shares bought with the dividend
Unit Price: price bought at with the dividend
Fee: X amount if applicable
Note
Tags

But then the USD/Euro amount at the bottom should **_not_** count towards the money you have invested in the account as, quite simply, you haven't invested any more money in the account (it's just self growing.) As for the fee....that _might_ need to be accounted for against the account? 

[sample.csv](https://github.com/ghostfolio/ghostfolio/files/10457278/sample.csv)


I would love to help but afraid I don't know TS. Thanks @dtslvr Tried to break this up into more digestible bits with the headings.

Edit- mucking around. Setting the dividend might work just for bookkeeping but I guess setting the BUY as the right amount of shares at price 0 seems to start accurately reflecting the true performance of a security holding.

**Reply by raskanskyz** [reply]
I'm having an issue with setting it as a "DIVIDEND" since the repurchased shares are not adding up in the portfolio

**Comment by raskanskyz** [comment]
here is another example from TDAmeritrade:

```
09/23/2022,QUALIFIED DIVIDEND (LMT),,LMT,,,14.50,,,,
09/23/2022,Bought 0.035 LMT @ 412.6419,0.035,LMT,412.6419,,-14.50,,,,
09/26/2022,ORDINARY DIVIDEND (SCHD),,SCHD,,,36.16,,,,
09/26/2022,Bought 0.533 SCHD @ 67.886,0.533,SCHD,67.886,,-36.16,,,,
```

as you can see, a DRIP consists of 2 records, the first indicates a recieved dividend and the following record is an ordinary "BUY" transaction.

**Comment by raskanskyz** [comment]
I started working on this feature, feel free to review!

https://github.com/ghostfolio/ghostfolio/pull/1963

**Comment by chrishas35** [comment]
Per feedback on #1963, I have a DIVIDEND and a BUY transaction with the buy having a price of $0.00. I'm seeing that each day one of these transactions occurs, the Overview has a dip in the graph. I suspect somehow the $0.00 price is being applied to all lots that security for that day rather than market data?

**Comment by gratefuluser23** [comment]
I wish I was able to provide some programming support as this is a feature I would really appreciate being implemented.
